### PR TITLE
f-account-info@v0.9.0 - Wired up the store to the template

### PR DESCRIPTION
**Pull request by oliversweb**
### Changed
- Wired up the new Vuex Store that exposes the State `consumer` to the Template plus wired up the load action
---
Wiring up the edit and save actions to follow on separate prs

### CODE REVIEW

File: packages/components/pages/f-account-info/package.json [hunk 1]
```diff
@@ -43,10 +43,11 @@
     "@justeat/f-globalisation": "1.0.0"
   },
   "peerDependencies": {
-    "@justeat/browserslist-config-fozzie": ">=1.2.0"
+    "@justeat/browserslist-config-fozzie": ">=1.2.0",
+    "vuex": "3.5.1"
```
Comment (DamianMullins) [inline packages/components/pages/f-account-info/package.json]: Peer deps should be set as a range, typically setting a minimum known version and allowing any version higher to allow for more flexibility in consuming applications.

```suggestion
    "vuex": ">=3.5.1"
```

File: packages/components/pages/f-account-info/src/components/AccountInfo.vue [hunk 2]
```diff
@@ -230,38 +221,44 @@ export default {
...
             }
         }
     },
 
-    mounted () {
+    created () {
+        if (!this.$store.hasModule('fAccountInfoModule')) {
+            this.$store.registerModule('fAccountInfoModule', fAccountInfoModule);
+        }
+    },
```
Comment (SachaZvetelman) [inline packages/components/pages/f-account-info/src/components/AccountInfo.vue:245]: Don't we normally do this in the consuming application? At least we are doing that for Checkout.

File: packages/components/pages/f-account-info/src/components/_tests/AccountInfo.test.js [hunk 3]
```diff
@@ -18,48 +36,106 @@ const createStore = ({ state = {}, actions = {} } = {}) => new Vuex.Store({
...
+    storeOverride = null,
+    initialiseOverride = null
+} = {}) => {
+    const store = storeOverride || createStore({ state, actions });
+    store.registerModule = registerStoreModuleSpy;
+    store.hasModule = hasModuleSpy;
+
+    if (initialiseOverride) {
+        AccountInfo.methods.initialise = initialiseOverride;
+    }
```
Comment (SachaZvetelman) [inline packages/components/pages/f-account-info/src/components/_tests/AccountInfo.test.js]: For this kind of pattern in JavaScript, I personally prefer this one-liner:
`AccountInfo.methods.initialise = initialiseOverride || AccountInfo.methods.initialise;`

File: packages/components/pages/f-account-info/src/components/AccountInfo.vue (hunk 2 above)
Comment (oliversweb) [reply packages/components/pages/f-account-info/src/components/AccountInfo.vue:245]: Yes, this is now the preferred way ... (_before it was thought before that we couldn't do this and used to do this in the parent (coreweb)_)

Comment (SachaZvetelman) [reply packages/components/pages/f-account-info/src/components/AccountInfo.vue:245]: Sure, as long as we always have a store, then I definitely prefer this approach 👍 
Just for my understanding, how are we ensuring we have a store when this component it used?

File: packages/components/pages/f-account-info/package.json [hunk 4]
```diff
@@ -43,10 +43,11 @@
     "@justeat/f-globalisation": "1.0.0"
   },
   "peerDependencies": {
-    "@justeat/browserslist-config-fozzie": ">=1.2.0"
+    "@justeat/browserslist-config-fozzie": ">=1.2.0",
+    "vuex": ">=3.5.1"
```
Comment (DamianMullins) [inline packages/components/pages/f-account-info/package.json:47]: I think you [might also need this change](https://github.com/justeat/fozzie-components/pull/1540/files#diff-8d69a0c8f25e9aa68bf113b42dee7ed1f1dacb2c71b1a292101d02863e631608) to further prevent the packages from being bundled up when built?

**Review by SachaZvetelman — APPROVED** [review]

**Review by DamianMullins — APPROVED** [review]

**Pull request merged**